### PR TITLE
add notification archetype and templates

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review

#### What are the relevant tickets?
https://github.com/mitodl/ocw-www/issues/69

#### What's this PR do?
We need a way to display a sticky notification at the top of every page in `ocw-www`, with the content being derived from content markdown.  This PR adds the `notification` archetype along with a template that will display the latest one at the top of every page.  Once a notification is dismissed, a `localStorage` key is set to remember that that specific notification was dismissed. 

#### How should this be manually tested?
 - Read the readme if you have never run `ocw-www` locally and spin the site up
 - You should see a notification with the text "Consider donating to OpenCourseWare to help further our efforts." (Don't dismiss it just yet)
 - Create a new notification with `cd site && hugo new notifications/test.md`, you should see the page refresh with an empty notification
 - Edit the content of the created file at `site/content/notifications/test.md` and save
 - Verify that the content shows up in the notification.
 - Dismiss the notification and refresh the page, the notification should not reappear
 - Delete the file at `site/content/notifications/test.md`, the site should refresh and show the give now notification
 - Dismiss the give now notification and refresh the page, it should not reappear

#### Out of scope
 - Clicking a link in the notification should dismiss the notification
 - Style bleed from `search.scss` is causing all links to render black, will create an issue for this and fix in a separate PR
 - Black header on the home page is style bleed from the search page caused by a regression in https://github.com/mitodl/ocw-www/pull/67

#### Screenshots (if appropriate)
image
image
image
image

### CODE REVIEW

**Comment by pdpinch** [comment]
Sorry, we had a deep divide over the banner color, but I think the orange outranked the blue. Can you update the colors to match this: https://projects.invisionapp.com/d/#/console/18609637/448296968/preview

File: site/content/notifications/givenow.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+id: "1617046894617923400"
+title: "Givenow"
+date: 2021-03-29T15:41:34-04:00
+headless: true
+---
+##### **Consider [donating to OpenCourseWare](https://giving.mit.edu/give/to/ocw/?utm_source=ocw&utm_medium=homepage_banner&utm_campaign=nextgen_home) to help further our efforts.**
```
Comment (pdpinch) [inline site/content/notifications/givenow.md:7]: We should use the text from https://docs.google.com/document/d/1FR3-IOhB2iSkQNR7QK7YIpL5ip4GVV3fFiHWGiWOQQU/edit

How will this work to link to the page in #71?

Comment (gumaerc) [reply site/content/notifications/givenow.md:7]: @pdpinch This was just temporary, as I was waiting for that PR to merge but it has now.  Linking to the other page would be as simple as adding a generic page with the markdown content in the doc, then linking to it with an absolute link.  Update incoming on that.

Comment (gumaerc) [reply site/content/notifications/givenow.md:7]: Although I should have indicated that with a comment...

**Comment by gumaerc** [comment]
@pdpinch screenshots updated with orange style

**Comment by gumaerc** [comment]
@pdpinch I updated the screenshots with beta site example copy and landing page.  Another question: should we close the notification upon clicking link text inside of it?

**Review by rhysyngsun — APPROVED** [review]
There's a lint error on the tests, but otherwise everything looks good. Feel free to merge once that's fixed and tests pass.

**Pull request merged**